include/alpaka/mem/view/AccessorAdaptor.hpp

```diff
@@ -0,0 +1,169 @@
+/* Copyright 2022 Andrea Bocci
+
+ * Permission to use, copy, modify, and/or distribute this software for any
+ * purpose with or without fee is hereby granted, provided that the above
+ * copyright notice and this permission notice appear in all copies.
+ *
+ * THE SOFTWARE IS PROVIDED “AS IS” AND ISC DISCLAIMS ALL WARRANTIES WITH
+ * REGARD TO THIS SOFTWARE INCLUDING ALL IMPLIED WARRANTIES OF
+ * MERCHANTABILITY AND FITNESS. IN NO EVENT SHALL ISC BE LIABLE FOR ANY
+ * SPECIAL, DIRECT, INDIRECT, OR CONSEQUENTIAL DAMAGES OR ANY DAMAGES
+ * WHATSOEVER RESULTING FROM LOSS OF USE, DATA OR PROFITS, WHETHER IN AN
+ * ACTION OF CONTRACT, NEGLIGENCE OR OTHER TORTIOUS ACTION, ARISING OUT OF OR
+ * IN CONNECTION WITH THE USE OR PERFORMANCE OF THIS SOFTWARE.
+ */
+
+#pragma once
+
+#include <alpaka/dim/Traits.hpp>
+#include <alpaka/extent/Traits.hpp>
+#include <alpaka/mem/view/ViewAccessor.hpp>
+
+#include <sstream>
+#include <type_traits>
+
+namespace alpaka
+{
+    namespace internal
+    {
+        template<typename TView>
+        struct AccessorAdaptor
+        {
+        private:
+            using value_type = Elem<TView>;
+            using pointer = value_type*;
+            using const_pointer = value_type const*;
+            using reference = value_type&;
+            using const_reference = value_type const&;
+
+        public:
+            AccessorAdaptor()
+            {
+                static_assert(experimental::traits::internal::IsView<TView>::value, "");
+            }
+
+            ALPAKA_FN_HOST auto data() -> pointer
+            {
+                return getPtrNative(*static_cast<TView*>(this));
+            }
+
+            ALPAKA_FN_HOST auto data() const -> const_pointer
+            {
+                return getPtrNative(*static_cast<TView const*>(this));
+            }
+
+            ALPAKA_FN_HOST auto operator*() -> reference
+            {
+                static_assert(
+                    alpaka::Dim<TView>::value == 0,
+                    "operator* is only valid for Buffers and Views of dimension 0");
+                return *data();
+            }
```

Comment:
I wonder whether we should SFINAE the operator away instead of static asserting. Opinions? Applies to all member functions with `static_assert`s on the dimensionality.

Reply:
I tried to do that, but I couldn't find a way to make it work.
The problem sees to that, due to the CRTP mixin approach, the type of `TView` is still incomplete at the time of the class definition, so it cannot be checked.
If you have suggestions for making it work, I can try to use the SFINAE approach instead of `static_assert`.

Reply:
You're right, you cannot get the dimensionality from `TView`. In LLAMA I had the same issue and went for an extra integral template parameter.

Reply:
Ok let's leave it as is for now.